plugins/pi4j-plugin-ffm/src/main/java/com/pi4j/plugin/ffm/providers/spi/FFMSpi.java

```diff
@@ -0,0 +1,135 @@
+package com.pi4j.plugin.ffm.providers.spi;
+
+import com.pi4j.context.Context;
+import com.pi4j.exception.InitializeException;
+import com.pi4j.io.exception.IOException;
+import com.pi4j.io.spi.*;
+
+public class FFMSpi extends SpiBase implements Spi {
+    protected static int SPI_BUS_MASK = 0x0100;
+    protected static int SPI_MODE_MASK = 0x0003;
+    protected static int SPI_WRITE_LSB_FIRST_MASK = 0x4000;
+    protected static int SPI_READ_LSB_FIRST_MASK = 0x8000;
+
+    public FFMSpi( SpiProvider provider, SpiConfig config) {
+        super(provider, config);
+
+        // get configured SPI bus
+        SpiBus bus = config.bus();
+
+        // get configured SPI mode
+        SpiMode mode = config.mode();
+
+        // the default value for 'flags' is zero
+        int flags = 0;
+
+        // if 'flags' were provided in the SPI config, then accept them
+        if(config().flags() != null){
+            flags = config().flags().intValue();
+        }
```

Comment:
in the linuxfs spi implementation we error out is the flags parm was used, This pertained to the pigpio provider and it continued use would put limits on the linuxfs code.